### ISSUE TITLE
desktop and window title fonts are disproportionally large after (dis)connecting a second monitor with different DPI

### ISSUE THREAD

**Issue by rathann** [mate-desktop/mate-control-center#261]
This was originally reported in Fedora 25 (regression from Fedora 24): https://bugzilla.redhat.com/show_bug.cgi?id=1397700 .

In System->Preferences->Look and Feel->Appearance->Fonts (`mate-appearance-properties -p fonts`), setting all fonts to size 10 makes the desktop and the window title fonts much larger than the rest at 96dpi and I have to set those two to 6 to make them appear the same size as the rest. At 160dpi, all fonts look more or less the same size.

MATE Desktop version 1.16 (I'm not sure which component is at fault here).
```
$ rpm -qa mate\*|sort
mate-applets-1.16.0-1.fc25.x86_64
mate-backgrounds-1.16.0-1.fc25.noarch
mate-calc-1.8.0-6.fc25.x86_64
mate-control-center-1.16.0-1.fc25.x86_64
mate-control-center-filesystem-1.16.0-1.fc25.x86_64
mate-desktop-1.16.1-1.fc25.x86_64
mate-desktop-libs-1.16.1-1.fc25.x86_64
mate-icon-theme-1.16.0-1.fc25.noarch
mate-icon-theme-faenza-1.16.0-1.fc25.noarch
mate-media-1.16.0-1.fc25.x86_64
mate-menus-1.16.0-1.fc25.x86_64
mate-menus-libs-1.16.0-1.fc25.x86_64
mate-menus-preferences-category-menu-1.16.0-1.fc25.x86_64
mate-notification-daemon-1.16.0-2.fc25.x86_64
mate-panel-1.16.0-2.fc25.x86_64
mate-panel-libs-1.16.0-2.fc25.x86_64
mate-polkit-1.16.0-1.fc25.x86_64
mate-power-manager-1.16.0-1.fc25.x86_64
mate-screensaver-1.16.0-1.fc25.x86_64
mate-screenshot-1.16.0-1.fc25.x86_64
mate-search-tool-1.16.0-1.fc25.x86_64
mate-sensors-applet-1.16.0-1.fc25.x86_64
mate-session-manager-1.16.0-1.fc25.x86_64
mate-settings-daemon-1.16.0-1.fc25.x86_64
mate-system-monitor-1.16.0-1.fc25.x86_64
mate-terminal-1.16.1-2.fc25.x86_64
mate-themes-3.22.4-1.fc25.noarch
mate-user-guide-1.16.0-1.fc25.noarch
mate-utils-common-1.16.0-1.fc25.noarch
```

Steps to reproduce:

1. Go to System->Preferences->Look and Feel->Appearance.
2. Select BlueMenta or TraditionalOk scheme (I have tested only these two).
3. Go to Fonts tab.
4. Click Details.
5. Set Resolution to 160dpi
6. Set all fonts' size to 8.
7. Connect an external monitor.
8. Set up to display only on the external monitor.
9. Set Resolution to 96dpi
10. Set all fonts' size to 10.

Actual results:
Desktop and Window Title fonts are larger than the rest.

Expected results:
All fonts should be the same size if I set them to the same size.

Additional info:
The reverse (Desktop and Window Title fonts too small) happens if I start with low-DPI screen (external monitor) and switch to a high-DPI screen (internal display).

Logging out and in "fixes" this.

My internal display is 13" 1920x1080 (~160DPI) and my external display is 24" 1920x1200 (~96DPI).

**Comment by rathann** [comment]
Still an issue with MATE 1.18.

**Comment by raveit65** [comment]
@vkareh 
Can you please take a look at this?
User said this isn't fixes with HIDPI support in 1.20 and fedora 28.
https://bugzilla.redhat.com/show_bug.cgi?id=1397700#c19

**Comment by vkareh** [comment]
Closing this and tracking in https://github.com/mate-desktop/marco/issues/493 and https://github.com/mate-desktop/caja/issues/1219
